core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -580,6 +605,27 @@ private synchronized void triggerUpdate()
         sendUpdate();
     }
 
+    /**
+     * Adaptively adjust batch size to meet expected request size:
+     * If requestSize is not equal to expectedSize, this function will try to estimate and adjust the batch size proportionally based on
+     * current nums of splits and size of request.
+     */
+    private synchronized void adjustSplitBatchSize(List<TaskSource> sources, long requestSize, long expectedSize)
+    {
+        int numSplits = 0;
+        for (TaskSource taskSource : sources) {
+            numSplits = Math.max(numSplits, taskSource.getSplits().size());
```

Comment:
With colocation joins with bucketed tables, there could be more than one partitioned sources. In that case, this logic will become awkward. Please apply this logic only when there is one partitioned source.